### PR TITLE
fix spelling errors

### PR DESCRIPTION
**Pull request by askeblad**
Cabal/src/Distribution/GetOpt.hs
cabal-install-solver/src/Distribution/Solver/Modular/Dependency.hs
cabal-install/src/Distribution/Client/FileMonitor.hs
cabal-install/src/Distribution/Client/GlobalFlags.hs
cabal-install/src/Distribution/Client/HttpUtils.hs
cabal-install/src/Distribution/Client/IndexUtils.hs
cabal-install/src/Distribution/Client/InstallPlan.hs
cabal-install/src/Distribution/Client/ProjectBuilding.hs
cabal-install/src/Distribution/Client/ProjectPlanning.hs
changelog.d/pr-7349
changelog.d/pr-7407

@Mikolaj missed these in `#8046`

<hr>

typo `concating` unattended to

cabal/Cabal/src/Distribution/Backpack/Id.hs

```
    -- show is found to be faster than intercalate and then replacement of
    -- special character used in intercalating. We cannot simply hash by
    -- doubly concating list, as it just flatten out the nested list, so
    -- different sources can produce same hash
```

typo `repocache` unattended to

cabal-install/src/Distribution/Client/IndexUtils.hs

```
    -- note that if this step fails due to a bad repocache, the the procedure can still succeed by reading from the existing cache, which is updated regardless.
```

### CODE REVIEW

**Review by jneira — APPROVED** [review]
thanks for correction

**Comment by Mikolaj** [comment]
@askeblad: oh, great. I think 'concating' should be 'concatenating' and `repocache` can probably be `repo cache`, unless we insist on inventing a new word.

**Comment by Mikolaj** [comment]
And Github Actions had a bad day, so don't worry.

**Comment by askeblad** [comment]
@grayjay care to make the second review

<hr>

> 'concating' should be 'concatenating' and `repocache` can probably be `repo cache`, 

@Mikolaj above changes might be added to a new PR
when `#7882` is resolved
`lossly` should be `loosely` or the "too obscure" lossily

**Comment by Mikolaj** [comment]
@askeblad: I don't follow. Would you like me to help you with #7882? E.g., I can make a decision how to fix the "lossly/loosely" conundrum, if that'd be helpful.

**Comment by askeblad** [comment]
on their own schedule `#7882` seems to be a matter for backpack authors to attend to 

at this point of time personally I don't want to push another commit associated with this PR

**Comment by Mikolaj** [comment]
@askeblad: absolutely, no pressure to push any more commits. I do appreciate the PR as is.

Regarding `#7882`, of course, if you prefer so, let's leave it be. But what exactly did you mean bringing it back? How is this related to this PR? I'm trying to understand. :) E.g., is my behaviour confusing? Do I do something wrong?

**Review by grayjay — APPROVED** [review]

**Comment by askeblad** [comment]
@Mikolaj excluding 4 .md file lines thus far all lines I've changed in this repo have been comment lines

`#7882` is another comment modification

**Comment by Mikolaj** [comment]
Yes. And? Please go on.

In the meantime, let me get this one merged. Thank you again!

**Comment by Mikolaj** [comment]
There is a technical problem: https://github.com/haskell/cabal/pull/8070/checks?check_run_id=5745374904

**Comment by askeblad** [comment]
>  technical problem

from ` #8046`

> help text linked

not that helpful

**Comment by askeblad** [comment]
you can run on a fork and merge

```
$ cat spelling_errors.sh
#! /bin/sh

sed -i 's/rewite/rewrite/g' cabal/cabal-install/src/Distribution/Client/FileMonitor.hs
sed -i 's/a-priori/a priori/g' cabal/cabal-install/src/Distribution/Client/GlobalFlags.hs
sed -i 's/checkin /checking /g' cabal/cabal-install/src/Distribution/Client/FileMonitor.hs
sed -i 's/dispay/display/g' cabal/cabal-install/src/Distribution/Client/HttpUtils.hs
sed -i 's/alterantively/alternatively/g' cabal/cabal-install/src/Distribution/Client/IndexUtils.hs
sed -i 's/packges/packages/g' cabal/cabal-install/src/Distribution/Client/InstallPlan.hs
sed -i 's/pre-emptively/preemptively/g' cabal/cabal-install/src/Distribution/Client/ProjectBuilding.hs
sed -i 's/Moreoever/Moreover/g' cabal/cabal-install/src/Distribution/Client/ProjectPlanning.hs
sed -i 's/conistently/consistently/g' cabal/cabal-install/src/Distribution/Client/ProjectPlanning.hs
sed -i 's/deppendencies/dependencies/g' cabal/cabal-install-solver/src/Distribution/Solver/Modular/Dependency.hs
sed -i 's/decription/description/g' cabal/Cabal/src/Distribution/GetOpt.hs
sed -i 's/decription/description/g' cabal/changelog.d/pr-7349
sed -i 's/decription/description/g' cabal/changelog.d/pr-7407
```
according to your discretion

```
$ sed -i 's/concating/concatenating/g' cabal/Cabal/src/Distribution/Backpack/Id.hs
$ sed -i 's/repocache/repo cache/g' cabal-install/src/Distribution/Client/IndexUtils.hs
$ 
```
according to Backpack authors' discretion

```
$ sed -i 's/lossly/loosely/g' cabal/Cabal-syntax/src/Distribution/Backpack.hs
$ 
```

**Comment by Mikolaj** [comment]
Thank you for the suggestion, but I'd rather do this the regular. Way. Is the recipe at

https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork

our of the question? Alternatively, you could rebase your branch manually if if we merged it prompty enough there'd be little risk of polluting ghi history with merges.

**Comment by Mikolaj** [comment]
@askeblad: you know what, never mind the complex git history, let's merge this as is.

**Comment by askeblad** [comment]
> rebase your branch manually 

@Mikolaj 

Anything you would change in the below (provided for reference)?

```sh
$ cd /tmp/cabal # clone of forked repo
$ git remote add upstream https://github.com/haskell/cabal.git
$ git fetch -q upstream
Successfully rebased and updated refs/heads/master.
$ 
```
Then force push to your forked repo with `git push -f origin master`.

**Comment by Mikolaj** [comment]
I think, after `git fetch upstream`, you also need `git rebase upstream/master`. The `Successfully rebased and updated refs/heads/master.` message probably only refers to various internally kept branches (e.g., `upstream/master`), not your own branch for this PR, even this is just `master`.

Otherwise, looks good.

**Comment by askeblad** [comment]
neglected to type the line when writing the comment

message was output by `git rebase upstream/master`

<hr>

don't know if repos `critbit`, `fgl`, `network-uri`, `old-locale` and `wreq`, are factually monitored

```
haskell/attoparsec README (update)
#204 opened 15 days

haskell/attoparsec use unicode
#203 opened 18 days ago

haskell/critbit fix typos
#96 opened 16 days ago

haskell/fgl fix typos
#101 opened 16 days ago

haskell/network-uri fix typo
#67 opened 15 days

haskell/old-locale update link
#2 opened 18 days ago

haskell/parsec fix typos
#145 opened 15 days ago

haskell/wreq fix typos
#146 opened 14 days ago
```

**Comment by askeblad** [comment]
> abandoned

if so might be moved to a newly created `github.com/haskell/pasture` repo directory

any outstanding questions on your end (if none don't reply)

eventually, _e.g._  Ed Yang, might get around to reviewing backpack comments

**Comment by Mikolaj** [comment]
Let's hope they all get around and also new maintainers arrive. I'm not a maintainer of any of these nor do I even have the right to these repos, so I can't help, I'm afraid.

**Comment by askeblad** [comment]
leaving this here for reference

quick and dirty comment extraction

```
$ git clone -q https://github.com/haskell/zlib.git`
$ cd zlib/test/Test/Codec/Compression/Zlib
$
```
grep

```
$ grep -n "^-- " *.hs
Internal.hs:3:-- | Test code and properties for "Codec.Compression.Zlib.Internal"
Stream.hs:3:-- | Test code and properties for "Codec.Compression.Zlib.Stream"
$ 
```
perl

```
$ perl -ne 'print if / -- /' *.hs
  -- GZipOrZlib omitted since it's not symmetric
                   -- These are disabled by default in the package
                   -- as they are only available with zlib >=1.2
                   -- ++ [RLE, Fixed]
$
```

**Comment by Mikolaj** [comment]
Nice trick!

**Comment by askeblad** [comment]
> backpack comments

@Mikolaj 

tempted to resubmit

this is probably correct

Cabal-syntax/src/Distribution/Backpack.hs

```diff
  -- In the absence of Backpack, there are no holes to fill, so any such
- -- component always has an empty module substitution; thus we can lossly
+ -- component always has an empty module substitution; thus we can lossily
  -- represent it as an 'OpenUnitId uid'.
```

leaving it for someone else to make a decision on `fgaz`'s suggestion

> Rather, "represent it as an 'OpenUnitId uid'" should be changed to "represent it as an 'DefiniteUnitId uid'"

**Comment by Mikolaj** [comment]
I have full confidence in @fgaz's judgement here so, if that helps, I hereby decide we can follow the suggestion and change to `DefiniteUnitId`. You can blame it on me. :)

Regardless, you are very welcome to submit a new PR, with one of both of the changes (or more).

**Comment by askeblad** [comment]
[Glossary](https://wiki.haskell.org/Category:Glossary) (Haskell Wiki)

 we have a _Vogonism_ entry fortunately

**Comment by askeblad** [comment]
@Mikolaj

are Block comments

start with the sequence `{-` and extend until the end sequence `-}`

very often encountered in Haskell code

network-uri/tests/uri001.hs

```
{- These (78-81) are some awkward test cases thrown up by a question on the URI list:
     http://lists.w3.org/Archives/Public/uri/2005Jul/0013
   Note that RFC 3986 discards path segments after the final '/' only when merging two
   paths - otherwise the final segment in the base URI is maintained.  This leads to
   difficulty in constructing a reversible relativeTo/relativeFrom pair of functions.
-}
```

**Comment by Mikolaj** [comment]
@askeblad: what about the block comments?

**Comment by askeblad** [comment]
if block comments in Haskell files are commonplace

might be useful to write a script to recursively extract them from a cloned repo

a sole block comment came to my attention in  Organization repos'

**Comment by Mikolaj** [comment]
Oh, I get it now. Indeed. they are commonplace for larger comments, I think.

**Comment by askeblad** [comment]
_cf_: https://github.com/haskell/cabal/pull/8070#issuecomment-1094336696

leaving this here for reference

quick and dirty locating of 1st block comment lines

```
$ grep -nr "^{- " network-uri
network-uri/tests/uri001.hs:645:{- These (78-81) are some awkward test cases thrown up by a question on the URI list:
$ 

```

**Comment by Mikolaj** [comment]
Thank you.

**Comment by askeblad** [comment]
@Mikolaj

Cabal/src/Distribution/GetOpt.hs

any reason for the pipe (first line) and backslash (third line)

```haskell
{-|
Process the command-line, and return the list of values that matched
(and those that didn\'t). The arguments are:
* The order requirements (see 'ArgOrder')
* The option descriptions (see 'OptDescr')
* The actual command line arguments (presumably got from
  'System.Environment.getArgs').
'getOpt' returns a triple consisting of the option arguments, a list
of non-options, and a list of error messages.
-}
```

**Comment by Mikolaj** [comment]
The pipe marks a haddock comment, which has its own form of a markup. I hope the backslash is not needed, though the markup uses the quote sign, as seen on later lines. So I would leave it be. If somebody renders this comment with `cabal haddock` and sees any errors, then it would be time to act.

**Pull request merged**
